### PR TITLE
kraken: improvement of disruptions application

### PR DESCRIPTION
**Pull request by kinnou02**
The circulation date of each circulation wasn't check, so it was
possible to disable a vj that wasn't impacted

### CODE REVIEW

**Comment by kinnou02** [comment]
over midnight case... yeah....

**Comment by antoine-de** [comment]
we (huuuum you :stuck_out_tongue_closed_eyes: ) should also test some stay_in case.

There are no reason for those to go well :smile:

**Comment by kinnou02** [comment]
kill me, kill me with fire :fire:

File: source/kraken/fill_disruption_from_chaos.cpp [hunk 1]
```diff
@@ -79,6 +81,21 @@ make_cause(const chaos::Cause& chaos_cause, nt::new_disruption::DisruptionHolder
...
+    auto day_period = bt::time_period(bt::ptime(date, bt::hours(0)), bt::hours(24));
+    //we get the validity period of the imapct for this day
+    return day_period.intersection(period);
+}
+
+//return the time period of circulation of a vj for one day
+static boost::posix_time::time_period execution_period(const boost::gregorian::date& date,
+                                                          const nt::VehicleJourney& vj){
+    return bt::time_period(bt::ptime(date, bt::seconds(vj.stop_time_list.front().departure_time)),
+            bt::ptime(date, bt::seconds(vj.stop_time_list.back().arrival_time)));
```
Comment (antoine-de) [inline source/kraken/fill_disruption_from_chaos.cpp]: I think you should add an epsilon to the arrival date, the end in not in the time_period

File: source/kraken/fill_disruption_from_chaos.cpp [hunk 2]
```diff
@@ -381,13 +414,17 @@ struct add_impacts_visitor : public apply_impacts_visitor {
             tmp_vp.days.set(i, vj.adapted_validity_pattern->days[i]);
```
Comment (antoine-de) [inline source/kraken/fill_disruption_from_chaos.cpp]: It really seems like a classic copy

**Pull request merged**